samples/ios/Continote/Continote/EditNoteViewController.swift

```diff
@@ -0,0 +1,98 @@
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import UIKit
+import Firebase
+import MaterialComponents.MaterialSnackbar
+
+/**
+ This ViewController presents the user with a Note editor to edit a specific Note, if
+ said Note is found for the user.
+ */
+class EditNoteViewController: BaseViewController {
+
+  // The key from the Firebase Realtime Database of the Note the user wants to edit.
+  // The Note the user wants to edit will be gathered from the Firebase Realtime Database once the
+  // user is confirmed to be signed in below. This is gathered separately on this screen,
+  // rather than passed from MyNotesViewController, to ensure the user is presented with the
+  // latest value for the Note (at the time of opening this screen).
+  var keyOfNoteToEdit: String!
+
+  // Firebase Realtime Database reference for the current user's Note to edit, based on the key.
+  private var noteToEditFirebaseDatabaseRef: DatabaseReference?
+
+  // UI outlets
+  @IBOutlet var temporaryTodoLabel: UILabel!
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    // Set up this screen's AppBar title.
+    title = Constants.Text.screenTitle
+
+    // Style all labels.
+    temporaryTodoLabel.applyAppTheme(for: .normalText)
+  }
+
+  override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+
+    guard keyOfNoteToEdit != nil else {
+      // The key for the Note to edit must be set for the user to be on this screen.
+      // This should never happen, but just in case.
+      navigationController?.popViewController(animated: true)
+      return
+    }
+  }
+
+  override func handleUserSignedIn(_ user: User) {
+    super.handleUserSignedIn(user)
+
+    // Try to get the current value of the Note to edit from the Firebase Realtime
+    // Database. We only get this value once here to keep things simple.
+    // The Note could change elsewhere while on this screen, thus saving could
+    // overwrite any of those changes.
+    noteToEditFirebaseDatabaseRef =
+      Database.database().reference().child("notes").child(user.uid).child(keyOfNoteToEdit)
+    noteToEditFirebaseDatabaseRef?.observeSingleEvent(of: .value, with: { (snapshot) in
+      if let noteToEdit = Note(with: snapshot) {
+        // TODO: Note to edit found, so pass it to a "handleNoteToEditFound(noteToEdit)" handler.
+      } else {
+        // TODO: Note to edit not found, call a "handleNoteToEditNotFound()" handler.
+      }
+    }) { (error) in
+      // TODO: Note to edit not found, call a "handleNoteToEditNotFound()" handler.
```

Comment:
The reason I said "Note _to edit_ not found" is because, technically, that's true for the current user - since they _cannot_ edit the note. Also, for this sample, to keep things simple I thought it would be fair to treat "Note not found" and "Cannot edit Note" the same (since the end result will be the user having to go back to the My Notes screen in either case, just like in the web sample).

However, that obviously wasn't clear, so I reworded this. Thanks!